homeassistant/components/spotify/media_player.py

```diff
@@ -137,6 +134,16 @@ def __init__(
         )
         self._currently_playing: dict | None = {}
         self._playlist: dict | None = None
+        self._restricted_device: bool = False
+
+    @property
+    def supported_features(self) -> MediaPlayerEntityFeature:
+        """Return the supported features."""
+        if self._restricted_device:
+            return MediaPlayerEntityFeature.SELECT_SOURCE
+        if self.data.current_user["product"] == "premium":
+            return SUPPORT_SPOTIFY
+        return MediaPlayerEntityFeature.SELECT_SOURCE
```

Comment:
Previously,. this was None, but I can't return None or nothing here. So unsure what to do here